src/internal/Subscriber.ts

```diff
@@ -171,15 +168,15 @@ export class SafeSubscriber<T> extends Subscriber<T> {
       error = (<PartialObserver<T>> observerOrNext).error;
       complete = (<PartialObserver<T>> observerOrNext).complete;
       if (observerOrNext !== emptyObserver) {
-        context = Object.create(observerOrNext);
+        next = next && next.bind(observerOrNext);
+        error = error && error.bind(observerOrNext);
+        complete = complete && complete.bind(observerOrNext);
         if (isSubscription(observerOrNext)) {
           observerOrNext.add(this.unsubscribe.bind(this));
         }
-        context.unsubscribe = this.unsubscribe.bind(this);
```

Comment:
I was able to remove this statement without affecting any tests; this makes me a bit unsure on what this was trying to achieve and if it's safe to remove it, although I can't think of what this was supposed to do.

Reply:
And I realized after opening this that without this assignment, it becomes possible to just use `context = observerOrNext` as an alternative to binding the subscriber callbacks. If that is indeed possible (again, no tests start to fail) then I think switching to use that approach makes sense.